Spanner/src/Transaction.php

```diff
@@ -615,6 +610,19 @@ public function commit(array $options = [])
             throw new \BadMethodCallException('The transaction cannot be committed because it is not active');
         }
 
+        // For commit, A transaction ID is mandatory for non-single-use transactions,
+        // and the `begin` option is not supported.
+        if (is_null($this->transactionId) && isset($this->options['begin'])) {
+            // Since the begin option is not supported in commit, unset it.
+            unset($this->options['begin']);
```

Comment:
I do have a question regarding this behavior... how is it that `begin` is set in a context where it is not supported? I feel like we should just let the API throw the appropriate error if a customer is supplying `begin` where they shouldn't be. If it's a matter of our testing, then we can fix our tests. But I'm confused why this is necessary.

Reply:
Consider the following scenario, A user calls `runTransaction` function like this:
```
database->runTransaction(function ($t) {
            $t->insert($table, $values);
            $t->commit();
        });
```

Here, `insert` is a mutation and it will get queued and processed with the `commit` operation. Since there's no server call except `commit`, ILB can't function. Therefore, the library must make a `beginTransaction` RPC before calling `commit`. This check is necessary because ILB is transparent to the user, and the `runTransaction` method automatically provides the `begin` option, unaware of the content in the user's callable.

Reply:
Thank you, that makes sense!